File: algorithms/ar-binsrt/java/binSort.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.Scanner;

public class Main {
  public void binSort(float[] arr, int n) {
    if (n <= 0)
      return;
    
    ArrayList<Float>[] bin = new ArrayList[n];

    
    for (int i = 0; i < n; i++)
      bin[i] = new ArrayList<Float>();

    for (int i = 0; i < n; i++) {
      int binIndex = (int) arr[i] * n;
      bin[binIndex].add(arr[i]);
    }

    
    for (int i = 0; i < n; i++) {
      Collections.sort((bin[i]));
    }

    
    int index = 0;
    for (int i = 0; i < n; i++) {
      for (int j = 0, size = bin[i].size(); j < size; j++) {
        arr[index++] = bin[i].get(j);
      }
    }
  }

  
  public static void main(String[] args) {
    Main b = new Main();  
    Scanner sc= new Scanner(System.in);
    System.out.println("Enter the no. of numbers");
    int n= sc.nextInt();
    float[] arr = new float[n];
    for(int i=0;i<n;i++)
    {   System.out.println("Enter the number");
        arr[i]=sc.nextFloat();
    }
    b.binSort(arr, n);

    for (float i : arr)
      System.out.print(i + "  ");
  }
}

File: algorithms/ar-binsrt/typescript/bin_sort.ts
var collection: Array<number> = new Array(0.897, 0.565, 0.656, 0.1234, 0.665, 0.3434);
var arrayList: number[][] = new Array<Array<number>>(collection.length);

function binSort(collection: Array<number>, size:number): void {
    for(let i=0; i<size; i++){
        arrayList[i] = new Array();
    }

    for(let i=0; i<size; i++){
        var index = Math.floor(collection[i]) * size;
        arrayList[index].push(collection[i]);
    }

    for(let element of arrayList){
        element.sort();
    }

    for(let i=0, index=0; i<size; i++){
        for(let j=0, arraySize=arrayList[i].length; j<arraySize; j++){
            collection[index++] = arrayList[i][j];
        }
    }
}

binSort(collection, collection.length);